### DISCUSSION TITLE
Custom Response Headers

### DISCUSSION THREAD

**Discussion by goll-michael** [microcks/microcks#1169] [Q&A] [answered]
Dear Community,

is there a way to define your own response headers in the operations?

Example: We would like to incorporate Api-Deprecations and communicate this in the Microcks contracts. For this purpose, a header “X-api-deprecation” with the value “true” should be returned.

Thanks in advance!

Best regards
Michael

**Comment by lbroudoux** [comment, marked as answer]
Hi @goll-michael,

Yes you can! Just integrate response headers in your example within your artifact. For example, if using an OpenAPI spec, you can have:

```yaml
    responses:
      '200':
        [...]
        headers:
          'X-api-deprecation':
            schema:
              type: boolean
            examples:
              example_1:
                value: true
              example_2:
                value: false
```

**Reply by goll-michael** [reply]
Hey @lbroudoux,

thank you very much for your quick reply. :-)
